.devcontainer/devcontainer.json

```
@@ -0,0 +1,48 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the README at:
+// https://github.com/microsoft/vscode-dev-containers/tree/v0.209.5/containers/python-3
+{
+	"name": "Python 3",
+	"build": {
+		"dockerfile": "Dockerfile",
+		"context": "..",
+		"args": {
+			// Update 'VARIANT' to pick a Python version: 3, 3.10, 3.9, 3.8, 3.7, 3.6
+			// Append -bullseye or -buster to pin to an OS version.
+			// Use -bullseye variants on local on arm64/Apple Silicon.
+			"VARIANT": "3.9-bullseye",
+		}
+	},
+
+	// Set *default* container specific settings.json values on container create.
+	"settings": { 
+		"python.defaultInterpreterPath": "/usr/local/bin/python",
+		"python.linting.enabled": true,
+		"python.linting.pylintEnabled": true,
```

```suggestion
		"python.linting.pylintEnabled": false,
```
pylintがImport Errorを起こすのでfalseにしたほうが良さそうです。
これをfalseにしても本当に存在しないimportに対する注意は正常に出ることを確認しています。

---

poetry で環境を作るので、コンテナ内に用意されているPythonやツールを使う必要も無さそうです。VSCodeでPythonインタープリタを指定すれば、poetryでインストールしたblackやflake8にパスが通るはずです。

また、KowerKointさんも指摘していますが、pylintを無効にしてflake8を有効にする、blackやisort、mypyの設定は最低限必要かと思っています。ここは、python-project-templateで使っているツールに合わせておくといいと思っています。

```json
	"settings": {
		"python.defaultInterpreterPath": "${containerWorkspaceFolder}/.venv/bin/python",
		"python.linting.enabled": true,
		"python.linting.pylintEnabled": false,
		"python.linting.flake8Enabled": true,
		"python.formatting.provider": "black",
		"editor.formatOnSave": true,
		"editor.codeActionsOnSave": {
			"source.organizeImports": true
		},
		"python.linting.mypyEnabled": true,
	},
```